pkg/model/resource/resource.go

```diff
@@ -40,6 +40,9 @@ type Resource struct {
 	// Controller specifies if a controller has been scaffolded.
 	Controller bool `json:"controller,omitempty"`
 
+	// Core indicates if this is a core resource.
+	Core bool `json:"core,omitempty"`
```

Comment:
I'm not sure why there are those json tags. Is it marshalled somewhere?

Reply:
Yep, in the [config](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/config/v3/config.go#L70). I think we can get the same functionality as setting a bool if we check `strings.HasPrefix(Resource.Path, modulePath)` where `modulePath` is from go.mod. This might break for submodules though.

If we do decide to go with an extra field, I vote for
```suggestion
	External bool `json:"external,omitempty"`
```
Since it covers both cases. If the path starts with `k8s.io/` then we can be pretty sure it is core.

Reply:
Thanks, thinking of that there is no point of introducing `.Core` because we can basically check `strings.HasPrefix(res.Path, "k8s.io")`.

Checking against module is also not hard. There is `golang.findGoModulePath` function already defined. It could be exported.

Reply:
@estroz I went with simple prefix check. That can be changed if for any reason `.External` has to be introduced, but I'd prefer to avoid changing `PROJECT` because something _may_ be needed.

Reply:
We cannot change core to external.
1) it is a breaking change
2) Core type means types that are defined in the K8s API and External means we scaffold an API that is defined in another project. For example, I have an Operator A which defines APIs that will also be used by my Operator B.